tsdb/docs/usage.md

```diff
@@ -0,0 +1,152 @@
+# Usage
+
+TSDB can be - and is - used by other applications such as [Cortex](https://cortexmetrics.io/) and [Thanos](https://thanos.io/).
+This directory contains documentation for any developers who wish to work on or with TSDB.
+
+Tip: `tsdb/db_test.go` demonstrates various usages of the TSDB library.
+
+## Instantiating a database
+
+Callers should use [`tsdb.Open`](https://pkg.go.dev/github.com/prometheus/prometheus/tsdb#Open) to open a TSDB.
+(the directory may be new or pre-existing).
+This returns a [`*tsdb.DB`](https://pkg.go.dev/github.com/prometheus/prometheus/tsdb#DB) which is the actual database.
+
+a `DB` has the following main components:
+
+* Compactor: a [leveled compactor](https://github.com/prometheus/prometheus/blob/c0c22ed04200a8d24d1d5719f605c85710f0d008/tsdb/compact.go#L78). Note: it is currently the only compactor implementation. It runs automatically.
+* [`Head`](https://pkg.go.dev/github.com/prometheus/prometheus/tsdb#DB.Head)
```

Comment:
There are also persistent blocks. You could point it to https://pkg.go.dev/github.com/prometheus/prometheus/tsdb#DB.Blocks.

Also, what was the reason to point leveled compactor it to the codebase and not https://pkg.go.dev/github.com/prometheus/prometheus/tsdb#LeveledCompactor?

Reply:
> Also, what was the reason to point leveled compactor it to the codebase and not https://pkg.go.dev/github.com/prometheus/prometheus/tsdb#LeveledCompactor?

no good reason.  until we decide on how to address the godoc problem, i have them a bit intermixed.